### PR TITLE
Release driver 1.1 and helm chart 1.1 (without dynamic provisioning)

### PR DESCRIPTION
**Pull request by wongma7**
**Is this a bug fix or adding new feature?**

Have decided that dynamic provisioning feature is not ready for release after its merge, as there are known issues being fixed like timeout happening if volume mount happens too soon after creation https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/303 plus security review. But nodegetvolumestats (merged months ago) and bottlerocket fix (merged weeks ago) are. Since the bottlerocket fix merged after dynamic provisioning, some git magic is needed to achieve a release 1.1 without dynamic provisioning

Don't know the best way to do this given our immature branching/release strategy but for transparency, options explored were:
1. Base release-1.1 on master then revert the dynamic provisioning commit from release-1.1 <- **PR is doing this** bc imo it's simplest
2. Revert the dynamic provisioning commit from master then base release-1.1 on it then merge it again later when it's totally ready for release <- too much effort, want to keep it in master branch so we keep testing it
3. Rebase release-1.1 on master and drop the dynamic provisioning commit from release-1.1 <- diff between release-1.1 and master becomes too hard to read
3. Cherry-pick all non dynamic provisioning commits on release-1.0 to form release-1.1 or v1.0.1 <- too much effort
4. Wrap dynamic provisioning code in a feature gate <- too much effort

TODO plan after this merges, same as with any release:
- Tag v1.1.0-rc.0
- Test upgrading from v1.0.0 to v1.1.0-rc.0 works with
  - helm update
  - kustomize
-  Make fixes if necessary
- Tag v1.1.0
- Manually release helm chart v1.1.0
- Cherrypick the release commit (doc, Makefile, and helm chart/kustomize) to master branch

### CODE REVIEW

**Comment by wongma7** [comment]
/cc @kbasv 
/cc @chrishenzie 
FYI!

A release is overdue so i can't wait for all the dynamic provisioning follow up to merge. At the same time I don't want to go thru hassle of revert it from master branch then merging it back later. So I'm just going to create release-1.1 branch by reverting the dynamic provisioning PR.

**Comment by wongma7** [comment]
Note the target branch.

**Comment by wongma7** [comment]
Travis seems to be confusing this PR with 305?? I might  have to reopen it.

File: CHANGELOG-1.x.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+# v1.1.0
+
+## Notable changes
+- The hostPath directory where the driver DaemonSet Pods write state files to their respective Node hosts has changed to fix the driver not working on Bottlerocket OS Nodes. No matter what OS your Nodes are running, you must use a supported method like helm or kustomize to update the driver. If not, i.e. if you only change the image tag of your DaemonSet, the migration from old to new directory won't succeed. See "change config dir location" below for details.
```
Comment (wongma7) [inline CHANGELOG-1.x.md:4]: Release notes were trimmed to remove all helm chart related changes which technically were already released in helm chart 1.0.0 last week.

And then I added this warning about not using helm/kustomize.

**Comment by nckturner** [comment]
> Base release-1.1 on master then revert the dynamic provisioning commit from release-1.1 <- PR is doing this bc imo it's simplest

I think this is a reasonable approach.

**Comment by kbasv** [comment]
Option 1 sounds good

**Comment by jqmichael** [comment]
I think another option is to follow what EBS CSI driver does to maintain [an overlay for alpha features](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/master/deploy/kubernetes/overlays/alpha). 

Not sure how that would work out with helm though.

/lgtm

**Comment by wongma7** [comment]
helm test:

```
# install 1.0.0
$helm install aws-efs-csi-driver aws-efs-csi-driver/aws-efs-csi-driver

#
$git checkout release-1.1

# edit static provisioning example so it has my fsid
$nvim examples/kubernetes/static_provisioning/specs/pv.yaml 

# create static provisioning example
$k create -f examples/kubernetes/static_provisioning/specs/

# check pod can still write
$k exec -it efs-app sh
sh-4.4# cat /data/out.txt
Thu Jan 28 18:57:00 UTC 2021
Thu Jan 28 18:57:05 UTC 2021
Thu Jan 28 18:57:10 UTC 2021
Thu Jan 28 18:57:15 UTC 2021

# upgrade with helm
$helm upgrade aws-efs-csi-driver ./helm --set image.tag=v1.1.0-rc.0
$k get po -n kube-system  efs-csi-node-zgvlm -o jsonpath="{.spec.containers[0].image}"
amazon/aws-efs-csi-driver:v1.1.0-rc.0
$k logs efs-csi-node-zgvlm -n kube-system efs-plugin
I0128 19:00:07.802365       1 config_dir.go:83] Pre-existing config files are being used from '/etc/amazon/efs-legacy'

# check pod can still write
sh-4.4# tail /data/out.txt 
Thu Jan 28 19:05:46 UTC 2021
Thu Jan 28 19:05:51 UTC 2021
Thu Jan 28 19:05:56 UTC 2021
Thu Jan 28 19:06:01 UTC 2021
Thu Jan 28 19:06:06 UTC 2021
```

Testing kustomize next!

**Comment by wongma7** [comment]
```
# install 1.0.0
$kubectl apply -k "github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/?ref=release-1.0"

# create static provisioning example
$k create -f examples/kubernetes/static_provisioning/specs/

# check pod can write
$k exec -it efs-app sh
sh-4.4# tail /data/out.txt 
Thu Jan 28 19:43:23 UTC 2021
Thu Jan 28 19:43:28 UTC 2021
Thu Jan 28 19:43:33 UTC 2021
Thu Jan 28 19:43:38 UTC 2021

# upgrade with overlay set to v1.1.0-rc.0
$kubectl apply -k ./deploy/kubernetes/overlays/stable
$k get po -n kube-system  efs-csi-node-s4r7g -o jsonpath="{.spec.containers[0].image}"
amazon/aws-efs-csi-driver:v1.1.0-rc.0^J
$k logs efs-csi-node-s4r7g -n kube-system efs-plugin
I0128 19:44:47.402451       1 config_dir.go:83] Pre-existing config files are being used from '/etc/amazon/efs-legacy'

# check pod can still write
sh-4.4# tail /data/out.txt 
Thu Jan 28 19:47:41 UTC 2021
Thu Jan 28 19:47:46 UTC 2021
Thu Jan 28 19:47:51 UTC 2021
Thu Jan 28 19:47:56 UTC 2021

**Pull request merged**